public/scripts/app.js

```diff
@@ -2,6 +2,171 @@ console.log("Sanity Check: JS is working!");
-
+//get all birds from db after the page reload
+$.ajax({
+  method: 'GET',
+  url: '/api/birds',
+})
+.then(function(birds) {
+  if (birds.length !== 0)
+    renderAllBirds(birds);
 });
+
+//adding a bird
+$("#bird-form form").on("submit", handleAddBird);
+//handling edit button click for editing a bird
+$("#birds").on("click", ".edit-bird", handelEditBird);
+//handling save button click for editing a bird
+$("#birds").on("click", ".save-bird", handleSaveBird);
+//handling delete button of a bird
+$("#birds").on("click", ".delete-bird", handleDeleteBird);
+
+function renderAllBirds(birds) {
+  birds.forEach(renderBird);
+}
+
+function renderBird(bird) {
+  console.log("entering renderBird()");
+
+  let htmlElem = `
+  <div class="panel panel-default bird" data-bird-id=${bird._id}>
+    <div class="panel-body">
+      <div class="row">
+        <div class="col-lg-3 col-md-6 col-sm-6 col-xs-12 image-container">
+          <img class="image" src="../../${bird.photo1}" alt="">
+        </div>
+
+        <div class="col-lg-3 col-md-6 col-sm-6 col-xs-12 image-container">
+          <img class="image" src="../../${bird.photo2}" alt="">
+        </div>
+
+        <div class="col-lg-6 col-md-12 col-sm-12 col-xs-12 info-container">
+          <ul class="list-group">
+            <li class="list-group-item">
+              <h4 class='inline-header'>Bird Name:</h4>
+              <span class='bird-name'>${bird.name}</span>
+            </li>
+
+            <li class="list-group-item">
+              <h4 class='inline-header'>Bird Type:</h4>
+              <span class='bird-type'>${bird.type}</span>
+            </li>
+
+            <li class="list-group-item">
+              <h4 class='inline-header'>Comments:</h4>
+              <span class='bird-comments'>${bird.comments}</span>
+            </li>
+
+            <li class="list-group-item">
+              <h4 class="inline-header">Learn more:</h4>
+              <a href="${bird.url}">${bird.urlName}</a>
+            </li>
+          </ul>
+
+        </div>
+      </div>
+    </div>
+
+    <div class="panel-footer">
+      <button class="btn btn-info edit-bird">Edit Bird</button>
+      <button class="btn btn-success save-bird hidden">Save Changes</button>
+      <button class="btn btn-danger delete-bird">Delete Bird</button>
+    </div>
+  </div>
+      `;
+
+  $("#birds").append(htmlElem);
+} //function renderBird(bird) {
+
+function handleAddBird(e) {
+  e.preventDefault();
+  console.log("app.js received submit button event");
+  let birdData = $(this).serialize();
+  console.log(`birdData: ${birdData}`);
+
+  $.post("/api/birds", birdData, function(addedBird) {
+    console.log(`addedBird: ${addedBird}`);
+    renderBird(addedBird);
+
+    $("[data-bird-id=" + addedBird._id + "]")[0].scrollIntoView();
+  });
+  $(this).trigger("reset");
+}
+
+function handelEditBird(e) {
+  e.preventDefault();
+  let birdId = $(this).closest(".bird").data("bird-id");
+  let $bird = $(this).closest(".bird");
```

Comment:
These two lines do the same bunch of work, and you could put them in the opposite order to great effect!

```js
let $bird = $(this).closest(".bird");
let birdId = $bird.data("bird-id");
```